source/guidelines/authentication.rst

```diff
@@ -83,10 +83,76 @@ web sites and services; it also seamlessly interacts with
 Kerberos. More information and use cases can be found in the `official
 documentation <https://www.keycloak.org/documentation>`_.
 
+.. note:: By `internal infrastructure` we mean also the access to Open
+   Data that are already available to the Open Data Hub Team but not
+   yet to the Data Consumers, and therefore require authentication.
+
 Source code for both the authentication server and a few pre-cooked
 examples of applications configured to connect to it can be found in
 dedicated servers created by the Open Data Hub Team: the
 `authentication server
 <https://github.com/noi-techpark/authentication-server-examples>`_,
 and the `example applications
 <https://github.com/noi-techpark/authentication-server-examples>`_,
+
+Quick howto
+~~~~~~~~~~~
+
+In order to access the internal infrastructure, you need first to get
+a token, then use it together with the API. Both steps can be achieved
+using command-line tools, for a programmatic access to the date, which
+is the method shown here.
+
+.. rubric:: Request an access token
+
+In order to receive an access token, you need in advance to have
+credentials for the Open Data Hub. If you do not have them, please
+open a ticket on issues.opendatahub.bz.it or send an email to
+:email:`help@opendatahub.bz.it`.
+
+With your username and password
+(:strong:`my_username`. `strong:`my_password`), the access token is
+granted to you with the following call:
+ 
+.. code-block:: bash
+   :name: grant-token
+   :caption: Receiving an access topic
+
+   curl -X POST -L -H 'Content-Type:application/x-www-form-urlencoded' \
+   "https://auth.opendatahub.bz.it/auth/realms/noi/protocol/openid-connect/token" \
+   -d 'grant_type=password&username=my_username&password=my_password&client_id=your-client-id'
+
+   
+The last parameter, :literal:`client_id` should be set to
+:strong:`odh-mobility-v2`.
+
+Since the token expires after a given amount of time, it might prove
+necessary to refresh it, an action that can be done by replacing the
+parameters given in the query above with
+
+.. code-block::
+   :name: refresh-token
+   :caption: Refreshing the access token
+	  
+   curl -X POST -L -H 'Content-Type:application/x-www-form-urlencoded' \
+   "https://auth.opendatahub.bz.it/auth/realms/noi/protocol/openid-connect/token" \
+   -d 'grant_type=refresh_token&refresh_token=*****&client_id=your-client-id'
+
+Here, use the refresh token received from
+:numref:`grant-token` and the same :literal:`client_id`. 
+
+
+.. rubric:: Retrieve data with the token.
+
+Once you received the access token, it is easy to use it in actual
+requests. The following API call shows how to get all
+:strong:`sname`\s and :strong:`mvalue`\s from the VMS dataset:
+	    
+.. code-block::
+   :name: get-closed-data
+   :caption: Retrieving data with the access token
+
+   curl -X GET 'https://mobility.api.opendatahub.bz.it/v2/api/flat/VMS/*?select=sname,mvalue' \
```

Comment:
Sorry @Piiit, but if you tell me  "Add this chapter..." I copy its content as much similar as the original I can. That curl example is **verbatim** from the flight rules so I assume that call never worked?

Anyway: what you mean exactly by `please add "latest" to the end of the URL`? I am asking because I can interpret as either of the following:

`'https://mobility.api.opendatahub.bz.it/v2/api/flat/VMS/*/latest?select=sname,mvalue'`
`'https://mobility.api.opendatahub.bz.it/v2/api/flat/VMS/latest/*?select=sname,mvalue'`

Reply:
Fine, it is a new feature, that was added after the flightrule... The flightrule has also to be updated. I will do that... it is just a hint, that some features have been changed and that you could update it adhoc.

What I mean is `curl -X GET 'https://mobility.api.opendatahub.bz.it/v2/api/flat/VMS/*/latest?select=sname,mvalue'`